run/order.go

```diff
@@ -42,6 +45,21 @@ func Sort(root string, stacks []stack.S) ([]stack.S, string, error) {
 		Str("root", root).
 		Logger()
 
+	logger.Trace().Msg("Computes implicit hierarchical order.")
+
+	sort.Sort(stacks)
+	for _, stack := range stacks {
+		for _, other := range stacks {
+			if stack.Path() == other.Path() {
+				continue
+			}
+
+			if strings.HasPrefix(stack.Path(), other.Path()+string(os.PathSeparator)) {
+				other.SetImplicitBeforePath(stack.Path())
+			}
+		}
+	}
```

Comment:
```suggestion
	for _, parent := range stacks {
		for _, child := range stacks {
			if parent.Path() == child.Path() {
				continue
			}

			if strings.HasPrefix(parent.Path(), child.Path()+string(os.PathSeparator)) {
				child.SetImplicitBeforePath(parent.Path())
			}
		}
	}
```

They are not always parent/child, but we are looking for parent/child relationships and it was a little hard at first for me to get which was the parent/child, so maybe renaming would improve things.

Reply:
yes but as you said, calling them `parent` and `child` is not correct... they are just stacks of the same list being compared. Maybe calling them `s1` and `s2` improves?